pkg/endpoint/endpoint.go

```diff
@@ -1635,7 +1635,7 @@ type MetadataResolverCB func(ns, podName string) (pod *slim_corev1.Pod, _ []slim
 //
 // This assumes that after the initial successful resolution, other mechanisms
 // will handle updates (such as pkg/k8s/watchers informers).
-func (e *Endpoint) RunMetadataResolver(resolveMetadata MetadataResolverCB) {
+func (e *Endpoint) RunMetadataResolver(bwm bandwidth.Manager, resolveMetadata MetadataResolverCB) {
```

Comment:
Can we just add the bw manager interface as a member variable of *Endpoint instead of passing it around like this (or do we expect this to be passed different instances of bw manager?)?

Reply:
> Can we just add the bw manager interface as a member variable of *Endpoint

That is harder than you make it sound. The current change is 4 callsites to `RunMetadataResolver`. 

Endpoints are not yet modularized so we gave to plumb the bandwidth manager to 5 callsites of `createEndpoint` which in turn is called via 42 other callsites, a portion of which will likely also need to get the bandwidth manager. So we are looking at a likely a 50 to 150 LOC change to do so :smiling_face_with_tear:

Reply:
I see, thanks for the explanation, my suggestion is likely not worth the trouble.

Perhaps we can add a TODO for whomever modularizes endpoints/epm so we can clean up this function signature later on?